src/libs/DateUtils.ts

```diff
@@ -336,6 +337,22 @@ function getStatusUntilDate(inputDate: string): string {
     return translateLocal('statusPage.untilTime', {time: format(input, `${CONST.DATE.FNS_FORMAT_STRING} ${CONST.DATE.LOCAL_TIME_FORMAT}`)});
 }
 
+/**
+ * Get a date and format this date using the UTC timezone.
+ * @param datetime
+ * @param dateFormat
+ * @returns If the date is valid, returns the formatted date with the UTC timezone, otherwise returns an empty string.
+ */
+function formatWithUTCTimeZone(datetime: string, dateFormat: string = CONST.DATE.FNS_FORMAT_STRING) {
+    const date = new Date(datetime);
+
+    if (isValid(date)) {
+        return timezoneFormat(utcToZonedTime(date, 'UTC'), dateFormat);
```

Comment:
Hi @narefyev91 

Thank you for your suggestion. 

I tested it using the user timezone from the preferences, but the problem is by default JavaScript uses the browser timezone when creating a Date. 

If we use the preferred timezone in the method `utcToZonedTime`, and if this timezone is the same as the browser timezone, the Date will be created decreasing/increasing the time related to the browser timezone.

For example, let's say the `created` date in the transaction is "**2023-10-26**", and we have these timezones below:

- **America/Sao_Paulo** with UTC offset `-03:00` (DST)
- **Africa/Abidjan** with UTC offset `+00:00`
- **Europe/Vilnius** with UTC offset `+03:00` (DST)

In this table, you can see the result for each timezone formatted with the format `yyyy-MM-dd'T'HH:mm:ssXXX`:

Where: 

- The column "**Formatted result preferred**" has the formatted date using the preferred timezone.
- The column "**Formatted result UTC**" has the formatted date using the `UTC` timezone (how is it in my code).

| Browser timezone      | Preferred timezone     | Formatted result preferred  | Formatted result UTC        |
| --------------------- | ---------------------- | --------------------------- | --------------------------- |
| **America/Sao_Paulo** | **America/Sao_Paulo**  | `2023-10-25T21:00:00-03:00` | `2023-10-26T00:00:00-03:00` |
| **America/Sao_Paulo** | **Africa/Abidjan**     | `2023-10-26T00:00:00-03:00` | `2023-10-26T00:00:00-03:00` |
| **America/Sao_Paulo** | **Europe/Vilnius**     | `2023-10-26T03:00:00-03:00` | `2023-10-26T00:00:00-03:00` |

When using the preferred timezone **America/Sao_Paulo** or **Europe/Vilnius** the date was decreased/increased accordingly to the timezone's UTC offset. If you see the information `XXX` in the results is always the same `03:00` because of the browser timezone.

That's why in my opinion using the `UTC` timezone when formatting the `created` information is better, because we always have the correct date format, in this case, what is important to us is only the date part of the result and not the time.

What do you think about it?

Reply:
Maybe we can use https://www.npmjs.com/package/date-fns-tz this one to generally tell which time zone we would like to have?

Reply:
Sure, I've been using some methods from `date-fns-tz` in the `formatWithUTCTimeZone` method, but I'll take a look at other ways to use a timezone from user preferences, and I bring the results.

Thank you @narefyev91

Reply:
In my test the `created` date in the transaction is "**2023-10-26**", and we have these timezones below:

America/Sao_Paulo with UTC offset -03:00 (DST)
Africa/Abidjan with UTC offset +00:00
Europe/Vilnius with UTC offset +03:00 (DST)
America/St_Kitts with UTC offset -04:00 (DST)

For each test, I formatted the date using the two date formats:

- Short: `yyyy-MM-dd`
- Full: `yyyy-MM-dd'T'HH:mm:ssXXX`

The column "**Using UTC in short format**" is the result using the fixed timezone `UTC` (how is it in my code).

**Note**: my browser timezone is **America/Sao_Paulo**

## Test using the method formatInTimeZone 

Using the method [formatInTimeZone](https://www.npmjs.com/package/date-fns-tz#formatintimezone): formats the date in the target time zone regardless of the **system's local time zone**.

I changed the method `formatWithUTCTimeZone` with the code below:

```
// Use the date in the parameter to format the value
return formatInTimeZone(datetime, timezone.selected, dateFormat)
```

Results:

| Preferred timezone     | Short format | Full format                 | Using UTC in short format |
| ---------------------- | ------------ | --------------------------- | ------------------------- |
| **America/Sao_Paulo**  | `2023-10-26` | `2023-10-26T00:00:00-03:00` | `2023-10-26`              |
| **Africa/Abidjan**     | `2023-10-26` | `2023-10-26T03:00:00Z`      | `2023-10-26`              |
| **Europe/Vilnius**     | `2023-10-26` | `2023-10-26T06:00:00+03:00` | `2023-10-26`              |
| **America/St_Kitts**   | `2023-10-25` | `2023-10-25T23:00:00-04:00` | `2023-10-26`              |

## Test using the method zonedTimeToUtc 

Using the method [zonedTimeToUtc](https://www.npmjs.com/package/date-fns-tz#zonedTimeToUtc): returns a Date with the equivalent UTC.

I changed the method `formatWithUTCTimeZone` with the code below:

```
// Create the date
const formattedCreated = zonedTimeToUtc(datetime, timezone.selected);

// Format the date
return tzFormat(formattedCreated, dateFormat, { timeZone: timezone.selected });
```

Results:

| Preferred timezone     | Short format | Full format                 | Using UTC in short format |
| ---------------------- | ------------ | --------------------------- | ------------------------- |
| **America/Sao_Paulo**  | `2023-10-26` | `2023-10-26T00:00:00-03:00` | `2023-10-26`              |
| **Africa/Abidjan**     | `2023-10-25` | `2023-10-25T21:00:00-03:00` | `2023-10-26`              |
| **Europe/Vilnius**     | `2023-10-25` | `2023-10-25T18:00:00-03:00` | `2023-10-26`              |
| **America/St_Kitts**   | `2023-10-26` | `2023-10-26T01:00:00-03:00` | `2023-10-26`              |

## Test using the method utcToZonedTime

Using the method [utcToZonedTime](https://www.npmjs.com/package/date-fns-tz#utcToZonedTime): format as the local time of any time zone from a specific UTC time.

I changed the method `formatWithUTCTimeZone` with the code below:

```
// Create the date
const formattedCreated = utcToZonedTime(datetime, timezone.selected);

// Format the date
return tzFormat(formattedCreated, dateFormat, { timeZone: timezone.selected })
```

Results:

| Preferred timezone     | Short format | Full format                 | Using UTC in short format |
| ---------------------- | ------------ | --------------------------- | ------------------------- |
| **America/Sao_Paulo**  | `2023-10-26` | `2023-10-26T00:00:00-03:00` | `2023-10-26`              |
| **Africa/Abidjan**     | `2023-10-26` | `2023-10-26T03:00:00Z`      | `2023-10-26`              |
| **Europe/Vilnius**     | `2023-10-26` | `2023-10-26T06:00:00+03:00` | `2023-10-26`              |
| **America/St_Kitts**   | `2023-10-25` | `2023-10-25T23:00:00-04:00` | `2023-10-26`              |

## Test using DateUtils internal method getLocalDateFromDatetime

Using `DateUtils` internal method [getLocalDateFromDatetime](https://github.com/Expensify/App/blob/020ba402f5bc3cfc3b0c1f3a15c73fb867067860/src/libs/DateUtils.ts#L78): this method uses `utcToZonedTime` to create a new date, but there is a logic to remove the timezone prefix from the date.

I changed the method `formatWithUTCTimeZone` with the code below:

```
// Create the date
const formattedCreated = getLocalDateFromDatetime('en-US', datetime, timezone.selected);

// Format the date
return tzFormat(formattedCreated, dateFormat, { timeZone: timezone.selected });
```

Results:

| Preferred timezone     | Short format | Full format                 | Using UTC in short format |
| ---------------------- | ------------ | --------------------------- | ------------------------- |
| **America/Sao_Paulo**  | `2023-10-25` | `2023-10-25T21:00:00-03:00` | `2023-10-26`              |
| **Africa/Abidjan**     | `2023-10-26` | `2023-10-26T00:00:00Z`      | `2023-10-26`              |
| **Europe/Vilnius**     | `2023-10-26` | `2023-10-26T03:00:00+03:00` | `2023-10-26`              |
| **America/St_Kitts**   | `2023-10-25` | `2023-10-25T20:00:00-04:00` | `2023-10-26`              |

## Conclusion

After making all these tests above, I concluded that using the fixed `UTC` timezone will guarantee that we'll always have the date (`yyyy-MM-dd`) as expected because each method from `date-fns-tz` will generate and format a date regarding the user's timezone.

@narefyev91, could you take a look at these tests to validate that you're okay with continuing to use fixed `UTC` to format the date, please?